pkg/db/queries.sql

```diff
@@ -14,7 +14,22 @@ WHERE
 
 -- name: InsertGatewayEnvelope :execrows
 SELECT
-	insert_gateway_envelope(@originator_id, @sequence_id, @topic, @originator_envelope);
+	insert_gateway_envelope(@originator_id, @originator_sequence_id, @topic, @originator_envelope);
+
+-- name: SelectGatewayEnvelopes :many
+SELECT
+	*
+FROM
+	gateway_envelopes
+WHERE (sqlc.narg('topic')::BYTEA IS NULL
+	OR topic = @topic)
+AND (sqlc.narg('originator_node_id')::INT IS NULL
+	OR originator_node_id = @originator_node_id)
+AND (sqlc.narg('originator_sequence_id')::BIGINT IS NULL
```

Comment:
unrelated to the PR, but who generates the sequence ID?

Reply:
Great question - the originator does, this would be extracted from the `originator_sid` field of `UnsignedOriginatorEnvelope`.

As for how the originator allocates these ID's in the first place, it is created by the `BIGSERIAL id` field of the `staged_originator_envelope` table - so we're using a postgres sequence for this.